src/telegram_to_engagement_db/telegram_group_to_engagement_db.py

```diff
@@ -0,0 +1,234 @@
+
+from src.telegram_to_engagement_db.cache import TelegramGroupSyncCache
+
+log = Logger(__name__)
+
+#TODO: move to social media tools?
+async def _initialize_telegram_client(telegram_token_file_url, google_cloud_credentials_file_path, pipeline_name):
+    """
+    :param telegram_token_file_url: Path to the Google Cloud file path that contains telegram tokens api_id
+                                                api_hash and telegram app admin phone number.
+    :type telegram_token_file_url: str
+    :param google_cloud_credentials_file_path: Path to the Google Cloud service account credentials file to use when
+                                               downloading facebook page token.
+    :type google_cloud_credentials_file_path: str
+    :param pipeline: Name of the pipeline for this telegram session.
+    :type pipeline: str
+    :return telegram_client
+    :rtype: telethon.client.telegramclient.TelegramClient
+
+    """
+    log.info('Downloading telegram access tokens...')
+    telegram_tokens = json.loads(google_cloud_utils.download_blob_to_string(
+        google_cloud_credentials_file_path, telegram_token_file_url).strip())
+
+    api_id = telegram_tokens['api_id']
+    api_hash = telegram_tokens['api_hash']
+    phone_number = telegram_tokens['phone_number']
+
+    log.info('Initializing telegram client...')
+    telegram = TelegramClient(f'{pipeline_name}_telegram_session_name', api_id, api_hash)
+    await telegram.start()
+
+    # Ensure the client is authorized
+    if await telegram.is_user_authorized() == False:
+        await telegram.send_code_request(phone_number)
+        try:
+            await telegram.sign_in(phone_number, input(f"Enter the authorization code sent to "
+                                                              f"telegram a/c for {phone_number}: "))
+        # If the a/c has 2FA enabled sign_in() will raise a SessionPasswordNeededError.
+        # Input your telegram password to proceed.
+        except SessionPasswordNeededError:
+            await telegram.sign_in(password=input('Password: '))
+    log.info('Initialized telegram client...')
+
+    return telegram
+
+
+async def _fetch_message_from_group(telegram, group_id, dataset_end_date=None, min_id=None):
+    """
+    :param telegram: Instance of telegram app to use to download the group messages from.
+    :type telegram: telethon.client.telegramclient.TelegramClient
+    :param group_id: Id of the telegram group to fetch messages from
+    :type group_id: str
+    :param dataset_end_date: Offset datetime, messages previous to this date will be retrieved. Exclusive
+    :type dataset_end_date: datetime | None
+    :param min_id: All the messages with a lower (older) ID or equal to this will be excluded.
+    :type min_id: int | None
+    :yields: Instances of telethon.tl.custom.message.Message
+    """
+    # Get group/channel entity
+    group_entity = await telegram.get_entity(PeerChannel(int(group_id)))
```

Comment:
Can group id be an int in the first place?

Reply:
Storing as a string so that we can reuse existing cache string values.

Reply:
Is it worth adding get_int and set_int to the cache? I'm slightly nervous if string/int conversions are going to cause us any problems, that's all.

Reply:
Updated to int in configurations. (Sorry, I had misread this as int conversion for start_message_id)